File: LeetCode/May challenge/Day 19/Day_19.py
class StockSpanner:

    def __init__(self):
        self.stack = []

    def next(self, price: int) -> int:
        count = 1
        while len(self.stack) > 0 and self.stack[-1][0] <= price:
            count += self.stack[-1][1]
            self.stack.pop()
        self.stack.append((price, count))
        return count

# Your StockSpanner object will be instantiated and called as such:
# obj = StockSpanner()
# param_1 = obj.next(price)

File: LeetCode/May challenge/Day 19/Day_19.cpp
class StockSpanner {
public:
    vector<pair<int, int>> vec;
    StockSpanner() {
        
    }
    
    int next(int price) {
        int count = 1;
        while(vec.size()>0 && vec[vec.size()-1].first<=price)
        {
            count+=vec[vec.size()-1].second;
            vec.pop_back();
        }
        vec.push_back(make_pair(price,count));
        
        return count;
    }
};

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner* obj = new StockSpanner();
 * int param_1 = obj->next(price);
 */
